web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,457 @@
+    const numPartitions = device.partitionTable.partitions.length;
+
+    return (
+      <>
+        <div>
+          {sprintf(n_("%d partition", "%d partitions", numPartitions), numPartitions)}
+        </div>
+        <div className="fs-small">
+          {sprintf(_("%s partition table"), device.partitionTable.type.toUpperCase())}
+        </div>
+      </>
+    );
+  };
+
+  const BlockContent = () => {
+    const systems = device.systems;
+    const filesystem = device.filesystem;
+
+    const content = () => {
+      if (systems.length > 0) return systems.join(", ");
+      if (device.filesystem?.isEFI) return _("EFI");
+
+      return _("Not identified");
+    };
+
+    return (
+      <>
+        <div>{content()}</div>
+        <If
+          condition={filesystem}
+          then={
+            <div className="fs-small">
+              {sprintf(_("%s file system"), filesystem?.type)}
+            </div>
+          }
+        />
+      </>
+    );
+  };
+
+  return (device.partitionTable ? <PartitionTableContent /> : <BlockContent />);
+};
+
+/**
+ * Column content with information about the size of the device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ */
+const DeviceSizeColumn = ({ device }) => {
+  const UnusedSize = () => {
+    const used = device.partitionTable?.partitions.reduce((s, p) => s + p.size, 0) || 0;
+    const unused = device.size - used;
+
+    if (unused === 0) return null;
+
+    return (
+      <div className="fs-small">
+        {sprintf(_("%s unused"), deviceSize(unused))}
+      </div>
+    );
+  };
+
+  const RecoverableSize = () => {
+    const size = device.recoverableSize;
+    let text;
+
+    if (size === 0)
+      text = _("No recoverable space");
+    else
+      text = sprintf(_("%s recoverable"), deviceSize(device.recoverableSize));
+
+    return <div className="fs-small">{text}</div>;
+  };
+
+  return (
+    <>
+      <div>{deviceSize(device.size)}</div>
+      <If condition={isDrive(device)} then={<UnusedSize />} else={<RecoverableSize /> } />
+    </>
+  );
+};
+
+/**
+ * Column content with the space action for a device.
+ * @component
+ *
+ * @param {object} props
+ * @param {StorageDevice} props.device
+ * @param {string} props.action - Possible values: "force_delete", "resize" or "keep".
+ * @param {boolean} [props.isDisabled=false]
+ * @param {(action: SpaceAction) => void} [props.onChange]
+ */
+const DeviceActionColumn = ({ device, action, isDisabled = false, onChange = noop }) => {
+  const changeAction = (_, action) => onChange({ device: device.name, action });
+
+  return (
+    <FormSelect
+      value={action}
+      isDisabled={isDisabled}
+      onChange={changeAction}
+      aria-label="Space action selector"
+    >
+      <FormSelectOption value="force_delete" label={_("Delete")} />
+      <FormSelectOption value="resize" label={_("Allow resize")} />
+      <FormSelectOption value="keep" label={_("Do not modify")} />
+    </FormSelect>
```

Comment:
Note for another iteration, and related with a discussion I had with @ancorgs yesterday: stop using form controls for actions that are going to be executed/applied right away on user interaction. This is a bit more difficult than exchanging a checkbox by a switch, but for sure we will find something.